server/save.js

```diff
@@ -17,7 +20,8 @@ exports.handleSave = function(req, res, app) {
   try {
     var user = res.locals.owner;
     var filename = utils.param(req, "file", utils.filenameFromUri(req));
-    var origfilename = filename;
+    var thumbname = path.join(path.dirname(filename), THUMB_DIR,
```

Comment:
@ulularem Nice suggestion! Should probably have something like `var absthumb = getAbsThumbPath(filename)`.